### PR TITLE
Session support for docs

### PR DESCRIPTION
**Pull request by vkarpov15**
Re: #6362 

There's two primary changes here:

1) Models now have a `startSession()` function that returns a promise that resolves to the underlying `ClientSession`. You can pass the `session` option to `Model.find()`, `Model.updateOne()`, etc.

2) Mongoose documents now have a `$session()` function that tracks the session associated with the document. If you load a document using `Model.findOne({}, null, { session })`, that document will track `session` and use that same session when you `save()` the document by default.

@mbroadst @daprahamian any thoughts you have would be much appreciated.

### CODE REVIEW

File: lib/model.js [hunk 1]
```diff
@@ -2122,6 +2128,41 @@ Model.watch = function(pipeline, options) {
   return new ChangeStream(this, pipeline, options);
 };
 
+/**
+ * _Requires MongoDB >= 3.6.0._ Starts a [MongoDB session](https://docs.mongodb.com/manual/release-notes/3.6/#client-sessions)
+ * for causal consistency in reads.
```
Comment (mbroadst) [inline lib/model.js]: I think probably better to leave it up to users to read up on why session are beneficial. It's not just for causal, but also retryable writes and eventually transactions (likely other features).

File: lib/model.js [hunk 2]
```diff
@@ -2122,6 +2128,41 @@ Model.watch = function(pipeline, options) {
...
+ *
+ * This function does not trigger any middleware.
+ *
+ * ####Example:
+ *
+ *     // Need to set `causalConsistency: true` to turn on causal consistency
```
Comment (mbroadst) [inline lib/model.js]: `causalConsistency` is `true` by default in sessions

File: test/model.test.js [hunk 3]
```diff
@@ -4821,7 +4821,7 @@ describe('Model', function() {
         });
       });
 
-      it('watch() (gh-5964)', function() {
+      it.skip('watch() (gh-5964)', function() {
```
Comment (mbroadst) [inline test/model.test.js]: intentionally skipped?

Comment (vkarpov15) [reply test/model.test.js]: Yes for now. We don't test against replica sets on travis currently, that's a separate concern, so I added this skip in https://github.com/Automattic/mongoose/pull/6363/commits/b44367885fa4168db58b09f4d76b4a08dc0abdbd

Comment (mbroadst) [reply test/model.test.js]: :+1:

**Pull request merged**